src/extraction/FStar.Extraction.ML.Syntax.fsti

```diff
@@ -239,8 +239,8 @@ val apply_obj_repr :  mlexpr -> mlty -> mlexpr
 
 val ty_param_names (tys:list ty_param) : list string
 
-val push_unit (ts : mltyscheme) : mltyscheme
-val pop_unit (ts : mltyscheme) : mltyscheme
+val push_unit (eff:e_tag) (ts : mltyscheme) : mltyscheme
+val pop_unit (ts : mltyscheme) : e_tag & mltyscheme
```

Comment:
`push_unit` is called when replacing the type arguments of a function with a unit.

`pop_unit` is the converse, eliminating a polymorphic function with a type application where the type argument is extracted as unit.

We now record the effect of type application as an e_tag, i.e., Pure, Erasable, Impure